node/resolvers/search/index.ts

```diff
@@ -432,13 +432,31 @@ export const queries = {
       hideUnavailableItems: args.hideUnavailableItems,
     }
 
-    const result = await biggySearch.facets(biggyArgs)
+    const facetPromises = [biggySearch.facets(biggyArgs)]
+
+    const showCategoryTree = args.categoryTreeBehavior === 'show';
+    const categorySelectedFacets = args.selectedFacets.filter(facet => facet.key === 'c')
+
+    if (!fullText && showCategoryTree && categorySelectedFacets.length > 0) {
```

Comment:
categorySelectedFacets can be undefined?

Reply:
Good point. I gave a double check and no, it can't be undefined. The `filter` function will always return an array